### PR TITLE
Add test for String.head

### PR DESCRIPTION
**Pull request by Dilven**

### CODE REVIEW

File: __tests__/String/head.test.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { S, O } from '../..'
+
+ describe('head', () => {
+   it('returns None if the given string is empty.', () => {
+     expect(S.head('')).toEqual(O.None)
+   })
+   it('Returns Some(value) where value is the first character of the string', () => {
+     expect(S.head('head-test')).toEqual('h')
+   })
+ })
```
Comment (mobily) [inline __tests__/String/head.test.ts:10]: (and again 😅) could you please add a test with the `*` string as the first argument of `it`?

File: src/String/String.res [hunk 2]
```diff
@@ -14,7 +14,10 @@ export concat = (str, append) => Js.String2.concat(str, append)
 %comment(
   "Returns `Some(value)` where `value` is the first character of the string, or `None` if the given string is empty."
 )
-export head = str => Js.String2.slice(str, ~from=0, ~to_=1)
+export head = str => {
+  let firstChar = Js.String2.slice(str, ~from=0, ~to_=1)
+  firstChar == "" ? None : Some(firstChar)
+}
```
Comment (mobily) [inline src/String/String.res:20]: let's reuse the `get` function :) 

```rescript
export head = str => get(str, 0)
```

by the way, nice catch! 💪

File: __tests__/String/head.test.ts (hunk 1 above)
Comment (Dilven) [reply __tests__/String/head.test.ts:10]: 😅

File: src/String/String.res (hunk 2 above)
Comment (Dilven) [reply src/String/String.res:20]: Great, I was wondering if I should use `get`

**Review by mobily — APPROVED** [review]
perfect, thank you! 🚀

**Pull request merged**